### PR TITLE
[OT210-44] Adding Delete Category

### PR DESCRIPTION
**Pull request by RodrigoMdza**
### Motivation and Summary

COMO usuario administrador
QUIERO eliminar una categoría existente
PARA mantener la información actualizada

Criterios de aceptación:
DELETE /categories/:id - Deberá validar que la categoría existe y eliminarla.

### Type of change
- [x] New feature
- [ ] New Tests
- [ ] Bug fix
- [ ] Refactor

### Checklist

- [x] Traceability between this change and Jira.
- [ ] ```mvn clean install``` was run and all tests completed successfully.
- [ ] There are no unused variables and/or imports in the modified classes.
- [ ] There are no imports in the modified classes with the wildcard character. Ex: ```com.somepackage.*```.
- [ ] Slf4j was used for the application log instead ```System.out```.
- [ ] Public methods are documented with ```javadoc```.

### CODE REVIEW

**Comment by manerajona** [comment]
@RodrigoMdza tiene conflictos con main

File: src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java [hunk 1]
```diff
@@ -19,4 +19,9 @@ public Long createEntity(Category category) {
         return categoryJpaRepository.save(category).getId();
     }
 
+    @Override
```
Comment (manerajona) [inline src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java:22]: Por favor agregar al método la anotación `@Transactional`

File: src/main/java/com/alkemy/ong/ports/input/rs/controller/CategoryController.java [hunk 2]
```diff
@@ -38,4 +37,12 @@ public ResponseEntity<Void> createCategory(@Valid @RequestBody CreateCategoryReq
                 .toUri();
         return ResponseEntity.created(location).build();
     }
+
+    @Override
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteCategory(@NotNull @PathVariable Long id) {
+        service.deleteCategory(id);
+        return new ResponseEntity<>(HttpStatus.OK);
```
Comment (manerajona) [inline src/main/java/com/alkemy/ong/ports/input/rs/controller/CategoryController.java:45]: Por favor retornar 204 NO CONTENT

**Review by manerajona — CHANGES_REQUESTED** [review]

File: src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -19,4 +19,9 @@ public Long createEntity(Category category) {
...
+    public void deleteCategory(Long id) {
+        categoryJpaRepository.deleteById(id);
```
Comment (manerajona) [inline src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java:24]: Esta tirando error 500 si se pasa un id que no existe, seguir el ejemplo: https://github.com/alkemyTech/OT210-server/blob/main/src/main/java/com/alkemy/ong/domain/usecase/impl/AlkymerServiceImpl.java#L62

image

**Review by manerajona — CHANGES_REQUESTED** [review]

File: src/main/java/com/alkemy/ong/ports/input/rs/controller/CategoryController.java (hunk 2 above)
Comment (RodrigoMdza) [reply src/main/java/com/alkemy/ong/ports/input/rs/controller/CategoryController.java]: Corregido

File: src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java (hunk 3 above)
Comment (RodrigoMdza) [reply src/main/java/com/alkemy/ong/domain/usecase/impl/CategoryServiceImpl.java]: Corregido

**Pull request merged**